File: MATLAB/AllTests/bubbleSort.m
function y = bubbleSort(x)

% 1. Compare adjacent items
% 2. swap pair if the items are out of order
% 3. repeat until at the end of the array
% 4. repaet steps 1-3 until array is sorted

%Note, this is slow: O(N^2) becuase of nested loops.

%boolean to determine if sort is complete or not
disordered = true;
y=x;
while(disordered)
    
    swapCount = 0;                                                          %init swapcount 
    
    for i = 1:(length(y))-1                                                 %check each pair of consequtive elements in turn
        if y(i) > y(i+1)
               y = swapElements(y, i, i+1);
               swapCount = swapCount + 1;                                   %increment swapCount if swap occurs
        end
    end
    
    if (swapCount == 0)                                                     %If no swaps occured, vector is ordered and loop condition set to false
        disordered = false;
    end
end

end

File: cpp/sortingAlgorithms/bubbleSort/src/supportingFiles/bubbleSort.cpp
#include "bubbleSort.h"


void bubbleSort(int arr[], int n)                                               //pass in array and array size
{
    int i, j;
    for (i = 0; i < n - 1; i++)

        for (j = 0; j < n - i - 1; j++)                                         //check each pair of consequetive elements in turn
            if (arr[j] > arr[j + 1])
                swap(&arr[j], &arr[j + 1]);
}